javascript.yml

```diff
@@ -0,0 +1,39 @@
+# Since the jobs have their stages defined here, makes sense for this code to own what stages can exist.
+stages:
+  - lint
+  - build
+  - unitTest
+  - deploy
+
+
+.js_lint:
+  stage: lint
+  variables:
+    eslint_version: "6"
+  script:
+    - docker run --rm -v $(pwd):/data cytopia/eslint:$eslint_version .
+
+.js_build:
+  stage: build
+  variables:
+    node_version: "13.12.0"
+  script:
+    - docker run --user $(id -u):$(id -g) --rm -w="/app" -v $(pwd):/app node:$node_version yarn
```

Comment:
what uid/giud is this typically?

Reply:
It should be gitlab-runners's, which seems to be 1001.  yarn was generating node modules as root, so this is a work around for that.